pkg/capabilities/triggers/on_demand_trigger.go

```diff
@@ -18,50 +19,69 @@ var info = capabilities.MustNewCapabilityInfo(
 type workflowID string
 
 type OnDemand struct {
+	log logger.Logger
 	capabilities.CapabilityInfo
 	chans map[workflowID]chan<- capabilities.CapabilityResponse
 	mu    sync.Mutex
+
+	// TO REVIEW: Need to ensure that slow consumer does not block this service as most method lock the services mutex
+	// including the send methods. We use a buffer to achieve this and drop (with logging) client messages if the buffer is full.
+	sendChannelBufferSize int
```

Comment:
@ettec You can drop this optimization for the trigger IMO -- this is really only meant as a example; we wouldn't ever use this in production.

Reply:
Discussed offline: we'll leave this in but remove the comment